addons/web/static/src/core/autocomplete/autocomplete.js

```diff
@@ -0,0 +1,324 @@
+/** @odoo-module **/
+
+import { useService } from "@web/core/utils/hooks";
+import { debounce } from "@web/core/utils/timing";
+import { getActiveHotkey } from "@web/core/hotkeys/hotkey_service";
+import { usePosition } from "@web/core/position_hook";
+
+const { Component, onWillUnmount, useExternalListener, useRef, useState, useEffect } = owl;
+
+function useDebounced(callback, timeout) {
+    const debounced = debounce(callback, timeout);
```

Comment:
also, bind the callback to component here (with usecomponent), so the user of the hook does not have to do it